apisix/plugins/traffic-split.lua

```diff
@@ -233,6 +233,27 @@ function _M.access(conf, ctx)
     local match_passed = true
 
     for _, rule in ipairs(conf.rules) do
+        -- check if all upstream_ids are valid
+        if rule.weighted_upstreams then
+            for _, wupstream in ipairs(rule.weighted_upstreams) do
+                local upstream_id = wupstream.upstream_id
+                if upstream_id then
+                    local key = "/upstreams/" .. upstream_id
+                    local res, err = core.etcd.get(key)
```

Comment:
Can we use `upstream.get_by_id(upstream_id) instead of this logic?

Reply:
I agree, the current scenario does not cover standalone mode.